slsDetectorSoftware/src/Module.cpp

```diff
@@ -1679,27 +1678,23 @@ void Module::getBadChannels(const std::string &fname) const {
     }
     // receive badchannels
     int nch = -1;
-    std::vector<int> badchannels;
     client.Receive(&nch, sizeof(nch));
+    std::vector<int> badchannels(nch);
     if (nch > 0) {
-        int temp[nch];
-        memset(temp, 0, sizeof(temp));
-        client.Receive(temp, sizeof(temp));
-        badchannels.insert(badchannels.end(), &temp[0], &temp[nch]);
```

Comment:
do you understand why this was inserting backwards? I think this might have been a bug that we had before?

Reply:
Empty vector, so inserting at the end() which is also the start.